### ISSUE TITLE
deps should be completed before vars are attempt

### ISSUE THREAD

**Issue by mgale** [go-task/task#1139]
The shell calls in vars should be executed after all the deps tasks are completed.

- Task version: 3.24.0
- Operating System: Ubuntu 22.04

In this example a stat error is raised because ./helloword does not exist.
```
version: '3'

tasks:
  # Get helloworld tool
  get-binary:
    cmds:
      - wget -nv https://<download-path>/helloworld
      - chmod +x helloworld
    status:
      - test -f helloworld

  do-stuff:
    deps: [get-binary]
    vars:
      FOO_BAR:
        sh: ./helloworld {{.ENV_FOO}}
    cmds:
      - echo "foo bar"
```

Output
```
stat /home/mgale/workspace/helloword: no such file or directory
task: Command "./helloworld ... " failed: exit status 127
```

**Comment by JonZeolla** [comment]
I'm seeing the same and I think it makes sense for deps to run first.

`Task version: v3.33.1 (h1:JJSRANHH7RQrr5Z2CTvSnTH7iWlfBlKV2W2O0JiZoLk=)`

**Comment by pd93** [comment]
Closing this as a duplicate of #535.

There are a few reasons why it is not currently possible to make this change in Task. The main one being that `deps` themselves can contain variables which might be dynamic (and therefore need to be resolved first). Currently, variables are all resolved at the same time and so either you do this before `deps` so that you can use vars in the dep call or you do it after as you are asking. Changing this behaviour now would break Taskfiles.

The solution to this is probably lazy loading variables which would support both solutions, but that requires a large overhaul of how variables work. This is something that we are considering, but there is no timeline associated with it.
